src/java.base/windows/classes/sun/nio/ch/PipeImpl.java

```diff
@@ -162,28 +167,34 @@ public void run() {
                     try {
                         if (ssc != null)
                             ssc.close();
+                        if (sa instanceof UnixDomainSocketAddress uaddr) {
+                            Files.deleteIfExists(uaddr.getPath());
+                        }
                     } catch (IOException e2) {}
                 }
             }
         }
     }
 
     /**
-     * Creates a Pipe implementation that supports buffering.
+     * Creates a (TCP) Pipe implementation that supports buffering.
      */
     PipeImpl(SelectorProvider sp) throws IOException {
-        this(sp, true);
+        this(sp, true, false);
     }
 
     /**
-     * Creates Pipe implementation that supports optionally buffering.
+     * Creates Pipe implementation that supports optionally buffering
+     * and is TCP by default, but if Unix domain is supported and
+     * preferAfUnix is true, then Unix domain sockets are used.
      *
-     * @implNote Uses a loopback connection. When buffering is
-     * disabled then it sets TCP_NODELAY on the sink channel.
+     * @param buffering if false set TCP_NODELAY on TCP sockets
+     *
+     * @param preferAfUnix use Unix domain sockets if supported
      */
     @SuppressWarnings("removal")
-    PipeImpl(SelectorProvider sp, boolean buffering) throws IOException {
-        Initializer initializer = new Initializer(sp);
+    PipeImpl(SelectorProvider sp, boolean buffering, boolean preferAfUnix) throws IOException {
```

Comment:
I think prefer AF_UNIX dominates, buffering is secondary, so I think the parameters should be reversed to make it clearer at the use-sites.